Output n_constrained_dofs

roystgnr opened the pull request:
## Reason
Additional output of numbers of constrained DoFs, when those exist

## Design
MeshInfo and ConsoleUtils have additional output when DoF constraints exist.

## Impact
Additional screen output and MeshInfo output

Closes #20909

roystgnr committed:
Report num_dofs_constrained in MeshInfo

Refs #20909

roystgnr committed:
Add num_dofs_constrained to MeshInfo reporter test

roystgnr committed:
Print n_constrained_dofs when it's non-zero

This doesn't get triggered often, only when Moose is starting from an
already-adaptively-refined or an IGA-constrained mesh or is enforcing
periodic BCs, but when it's triggered it's useful information.

Refs #20909

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1031354/) on e6fcd6f wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/20910/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/20910/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 8c8e35d85055aba71f02ed67330dc9c70a7c3092`

roystgnr committed:
clang-format changes

Of course the lines I pasted and modified can't be formatted the same
way as the lines I cut.  What was I thinking!?

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1031554/) on d5c83dc wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/20910/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1031563/) on d5c83dc wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>62e10e</th>
      <th colspan=4>#20910 d5c83d</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>82.51%</td>
      <td>82.51%</td>
      <td>-0.00%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>74711</td>
      <td>74714</td>
      <td>+3</td>
      <td>7</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15833</td>
      <td>15837</td>
      <td>+4</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20910/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20910/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/20910/coverage/xfem)

</details>



This comment will be updated on new commits.
